File: Easy/runningSum.cpp
// Problem 1480 :o

#include<iostream>
#include<cstdlib>
using namespace std;

class Solution {
public:
  vector<int> runningSum(vector<int>& nums) {
    vector<int> ans(nums.size());
    ans[0] = nums[0];
    for (int i = 1; i < nums.size(); i++)
      ans[i] = ans[i-1] + nums[i];
    return ans;
    }
};

int main(){
  
  return 0;
};

File: Easy/runningSum.rs
// 1480

impl Solution {
  pub fn running_sum(nums: Vec<i32>) -> Vec<i32> {
    let vector_length = nums.len();
    let mut result = vec![0; vector_length];

    if vector_length == 0 {
      return result;
    }

    for i in 0..vector_length {
      if i == 0 {
        result[i] = nums[i];
      } else {
        result[i] = result[i - 1] + nums[i];
      }
    }

    result
  }
}